### PR TITLE
Remove forest-fil-types crate and relocate items

### LINKED ISSUE #2645: Refactor the verifier code in `forest_types`
**Issue summary**

The `forest_types` crate contains a single module with a bunch of verifier functions. These functions do not belong in `forest_types`. Figure out a better place to put them and then delete the `forest_types` crate.

**Other information and links**

### PR DESCRIPTION
**Pull request by creativcoder**
## Summary of changes

Changes introduced in this pull request:

- Removes forest-fil-types crate and moves items under `state_manager/utils.rs`

## Reference issue to close (if applicable)

Closes #2645 

## Change checklist

- [x] I have performed a self-review of my own code,
- [X] I have made corresponding changes to the documentation,
- [X] I have added tests that prove my fix is effective or that my feature works
      (if possible),
- [X] I have made sure the
      [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is
      up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: blockchain/state_manager/Cargo.toml [hunk 1]
```diff
@@ -14,11 +14,11 @@ byteorder.workspace = true
 cid.workspace = true
 fil_actor_interface.workspace = true
 fil_actors_runtime_v9.workspace = true
+filecoin-proofs-api = { workspace = true, default-features = false }
```
Comment (lemmih) [inline blockchain/state_manager/Cargo.toml:17]:
```suggestion
filecoin-proofs-api = { workspace = true }
```

**Review by lemmih — CHANGES_REQUESTED** [review]
Why do these functions belong with the state manager?

**Comment by creativcoder** [comment]
I saw various sector info and PoST proof related apis already in state_manager/utils.rs so felt it would be a good place.

**Comment by lemmih** [comment]
> > Why do these functions belong with the state manager?
> 
> I saw various sector info and PoST proof related apis already in state_manager/utils.rs so felt it would be a good place.

If the state manager was the only consumer of these functions then it would make sense to put them here. But you're exporting the functions and importing them elsewhere. I'm not sure it logically makes sense to associate them with the state manager.

File: blockchain/consensus/fil_cns/src/validation.rs [hunk 2]
```diff
@@ -388,10 +397,71 @@ fn verify_winning_post_proof<DB: Blockstore + Clone + Send + Sync + 'static>(
...
+/// miners that are elected to mine a new block to verify a sector. A failed
+/// winning proof leads to a miner being slashed.
+
+/// PoSt proof variants.
+enum ProofType {
+    Winning,
+    // Window,
+}
+
+pub fn verify_winning_post(
```
Comment (lemmih) [inline blockchain/consensus/fil_cns/src/validation.rs:440]:
```suggestion
fn verify_winning_post(
```

File: blockchain/state_manager/src/utils.rs [hunk 3]
```diff
@@ -151,6 +152,24 @@ pub fn is_valid_for_sending(network_version: NetworkVersion, actor: &ActorState)
     };
 }
 
+/// Generates sector challenge indexes for use in winning PoSt verification.
+pub fn generate_winning_post_sector_challenge(
```
Comment (lemmih) [inline blockchain/state_manager/src/utils.rs:156]:
```suggestion
fn generate_winning_post_sector_challenge(
```

**Review by lemmih — CHANGES_REQUESTED** [review]

File: blockchain/state_manager/src/utils.rs [hunk 4]
```diff
@@ -151,6 +152,24 @@ pub fn is_valid_for_sending(network_version: NetworkVersion, actor: &ActorState)
     };
 }
 
+/// Generates sector challenge indexes for use in winning PoSt verification.
+fn generate_winning_post_sector_challenge(
```
Comment (lemmih) [inline blockchain/state_manager/src/utils.rs:156]: Could this be inlined?

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**